### PR TITLE
Ft test admin 147010549

### PR DESCRIPTION
**Pull request by bkiseu**
### What does this PR do?
Worked on the api tests
### Description of Task to be completed? *
Added tests and asserts on the 10 api tests files to make the tests pass
### What are the relevant GitHub issues/pivotal tracker stories? 
Api tests story

### CODE REVIEW

File: hc/api/tests/test_badge.py [hunk 1]
```diff
@@ -14,11 +14,16 @@ def setUp(self):
     def test_it_rejects_bad_signature(self):
         r = self.client.get("/badge/%s/12345678/foo.svg" % self.alice.username)
         ### Assert the expected response status code
+        self.assertEqual(400, r.status_code, msg='should return 404 error on bad signature')
 
     def test_it_returns_svg(self):
         sig = base64_hmac(str(self.alice.username), "foo", settings.SECRET_KEY)
         sig = sig[:8].decode("utf-8")
+        print(sig)
```
Comment (bmwachajr) [inline hc/api/tests/test_badge.py]: why are you printing sig ?

Comment (domiebett) [inline hc/api/tests/test_badge.py]: Do you have to print this?

File: hc/api/tests/test_check_model.py [hunk 2]
```diff
@@ -36,3 +40,9 @@ def test_paused_check_is_not_in_grace_period(self):
         self.assertFalse(check.in_grace_period())
 
     ### Test that when a new check is created, it is not in the grace period
+
+    def test_new_not_in_grace_period(self):
```
Comment (bmwachajr) [inline hc/api/tests/test_check_model.py]: method is not cescriptive enough perhaps use
test_new_check_not_in_grace_period(self):

File: hc/api/tests/test_create_check.py [hunk 3]
```diff
@@ -46,24 +49,35 @@ def test_it_works(self):
...
 
     def test_it_accepts_api_key_in_header(self):
-        payload = json.dumps({"name": "Foo"})
-
+        payload = {"name": "Foo"}
         ### Make the post request and get the response
-        r = {'status_code': 201} ### This is just a placeholder variable
 
-        self.assertEqual(r['status_code'], 201)
+        # r = {'status_code': 201}
```
Comment (Farhanisto) [inline hc/api/tests/test_create_check.py]: remove commented code

File: hc/api/tests/test_badge.py [hunk 4]
```diff
@@ -14,11 +14,16 @@ def setUp(self):
     def test_it_rejects_bad_signature(self):
         r = self.client.get("/badge/%s/12345678/foo.svg" % self.alice.username)
         ### Assert the expected response status code
+        self.assertEqual(400, r.status_code, msg='should return 404 error on bad signature')
 
     def test_it_returns_svg(self):
         sig = base64_hmac(str(self.alice.username), "foo", settings.SECRET_KEY)
```
Comment (andela-landia) [inline hc/api/tests/test_badge.py]: could `sig` be named better i.e. more descriptive?

File: hc/api/tests/test_badge.py [hunk 5]
```diff
@@ -14,11 +14,16 @@ def setUp(self):
...
         sig = sig[:8].decode("utf-8")
+        print(sig)
         url = "/badge/%s/%s/foo.svg" % (self.alice.username, sig)
```
Comment (andela-landia) [inline hc/api/tests/test_badge.py]: consider using `format` instead of %

File: hc/api/tests/test_create_check.py [hunk 6]
```diff
@@ -46,24 +49,35 @@ def test_it_works(self):
...
     def test_it_handles_missing_request_body(self):
         ### Make the post request with a missing body and get the response
-        r = {'status_code': 400, 'error': "wrong api_key"} ### This is just a placeholder variable
-        self.assertEqual(r['status_code'], 400)
-        self.assertEqual(r["error"], "wrong api_key")
+        payload = {}
+        response = self.client.post(self.URL, json.dumps(payload),
+                                    content_type="application/json")
+
+        # r = {'status_code': 400, 'error': "wrong api_key"} ### This is just a placeholder variable
```
Comment (domiebett) [inline hc/api/tests/test_create_check.py]: Do you need this commented out line?

File: hc/api/tests/test_create_check.py [hunk 7]
```diff
@@ -12,11 +12,12 @@ def setUp(self):
 
     def post(self, data, expected_error=None):
         r = self.client.post(self.URL, json.dumps(data),
```
Comment (andela-landia) [inline hc/api/tests/test_create_check.py:14]: Improve on variable naming from r to either response or result

File: hc/api/tests/test_create_check.py [hunk 8]
```diff
@@ -46,24 +49,35 @@ def test_it_works(self):
         self.assertEqual(check.grace.total_seconds(), 60)
 
     def test_it_accepts_api_key_in_header(self):
-        payload = json.dumps({"name": "Foo"})
-
+        payload = {"name": "Foo"}
         ### Make the post request and get the response
-        r = {'status_code': 201} ### This is just a placeholder variable
```
Comment (andela-landia) [inline hc/api/tests/test_create_check.py:54]: Remove new line

File: hc/api/tests/test_create_check.py [hunk 9]
```diff
@@ -46,24 +49,35 @@ def test_it_works(self):
...
 
     def test_it_handles_invalid_json(self):
         ### Make the post request with invalid json data type
-        r = {'status_code': 400, 'error': "could not parse request body"} ### This is just a placeholder variable
-        self.assertEqual(r['status_code'], 400)
-        self.assertEqual(r["error"], "could not parse request body")
+        payload = {"name": "Foo"}
+        response = self.client.post(self.URL, payload,
+                                    content_type="application/json")
+        # r = {'status_code': 400, 'error': "could not parse request body"} ### This is just a placeholder variable
```
Comment (domiebett) [inline hc/api/tests/test_create_check.py]: Does this line have a purpose?

File: hc/api/tests/test_create_check.py [hunk 10]
```diff
@@ -46,24 +49,35 @@ def test_it_works(self):
...
         ### Make the post request and get the response
-        r = {'status_code': 201} ### This is just a placeholder variable
 
-        self.assertEqual(r['status_code'], 201)
+        # r = {'status_code': 201}
+        response = self.client.post(self.URL, json.dumps(payload), HTTP_X_API_KEY="abc",
+                             content_type="application/json")
+
+        ### This is just a placeholder variable
+
```
Comment (andela-landia) [inline hc/api/tests/test_create_check.py]: Remove new line

File: hc/api/tests/test_sendalerts.py [hunk 11]
```diff
@@ -30,13 +30,14 @@ def test_it_handles_few(self, mock):
...
         # 1 day 30 minutes after ping the check is in grace period:
         check.last_ping = timezone.now() - timedelta(days=1, minutes=30)
         check.save()
 
         # Expect no exceptions--
-        Command().handle_one(check)
-
+        result = Command().handle_one(check)
     ### Assert when Command's handle many that when handle_many should return True
+        assert result is True
```
Comment (bmwachajr) [inline hc/api/tests/test_sendalerts.py:43]: you asserted ommand's handle_one but you should have asserted command  handle_many instead

File: hc/api/tests/test_create_check.py [hunk 12]
```diff
@@ -78,4 +92,33 @@ def test_it_rejects_non_string_name(self):
...
+        })
+
+
+
+        check_channel = [channel for channel in Check.objects.get().channel_set.all()][0]
+
+        self.assertEqual(check_channel.value, "http://exampletest.com")
+
     ### Test for the 'timeout is too small' and 'timeout is too large' errors
+    def test_timout_is_too_small(self):
```
Comment (domiebett) [inline hc/api/tests/test_create_check.py]: typo in 'timeout'

File: hc/api/tests/test_list_checks.py [hunk 13]
```diff
@@ -32,25 +32,45 @@ def get(self):
         return self.client.get("/api/v1/checks/", HTTP_X_API_KEY="abc")
 
     def test_it_works(self):
-        r = self.get()
+        response = self.get()
```
Comment (andela-landia) [inline hc/api/tests/test_list_checks.py:35]: is it self.get or self.client.get?

File: hc/api/tests/test_ping.py [hunk 14]
```diff
@@ -63,11 +65,29 @@ def test_it_reads_forwarded_protocol(self):
...
+    def test_ping_to_post(self):
+        response = self.client.post("/ping/%s/" % self.check.code)
+        assert response.status_code == 200
+
     ### Test that the csrf_client head works
+    def test_the_csrf_client_head_works(self):
+        self.csrf_client = Client(enforce_csrf_checks=True)
+        self.assertTrue(self.csrf_client)
+        r = self.csrf_client.get("/ping/%s/" % self.check.code)
+        assert r.status_code == 200
```
Comment (bmwachajr) [inline hc/api/tests/test_ping.py]: assert contents of head

File: hc/api/tests/test_list_checks.py [hunk 15]
```diff
@@ -32,25 +32,45 @@ def get(self):
...
-        doc = r.json()
+        doc = response.json()
         self.assertTrue("checks" in doc)
 
         checks = {check["name"]: check for check in doc["checks"]}
         ### Assert the expected length of checks
+        self.assertEqual(len(checks), 2)
+
         ### Assert the checks Alice 1 and Alice 2's timeout, grace, ping_url, status,
+        print(checks)
```
Comment (andela-landia) [inline hc/api/tests/test_list_checks.py]: You do not need to print

File: hc/api/tests/test_ping.py [hunk 16]
```diff
@@ -63,11 +65,29 @@ def test_it_reads_forwarded_protocol(self):
...
+    def test_paused_ping_changes_status(self):
+        assert self.check.status == "new"
+        self.check.status = "paused"
+        self.client.get("/ping/%s/" % self.check.code)
+        self.check.refresh_from_db()
+        assert self.check.status == "up"
+
     ### Test that a post to a ping works
+    def test_ping_to_post(self):
+        response = self.client.post("/ping/%s/" % self.check.code)
```
Comment (andela-landia) [inline hc/api/tests/test_ping.py]: Use `format` instead of `%`

File: hc/api/tests/test_pause.py [hunk 17]
```diff
@@ -25,3 +27,7 @@ def test_it_validates_ownership(self):
         self.assertEqual(r.status_code, 400)
 
         ### Test that it only allows post requests
+        response = self.client.get(url, "", content_type="application/json",
+                             HTTP_X_API_KEY="abc")
+
+        self.assertEqual(response.status_code, 405)
```
Comment (bmwachajr) [inline hc/api/tests/test_pause.py:33]: maintain consistence in coding style
`assert response.status_code == 200` or ` self.assertEqual(response.status_code, 405)`
stick to one style of assertion

File: hc/api/tests/test_pause.py [hunk 18]
```diff
@@ -9,10 +9,12 @@ def test_it_works(self):
         check.save()
 
         url = "/api/v1/checks/%s/pause" % check.code
-        r = self.client.post(url, "", content_type="application/json",
+        response = self.client.post(url, "", content_type="application/json",
                              HTTP_X_API_KEY="abc")
 
         ### Assert the expected status code and check's status
+        assert response.status_code == 200
```
Comment (bmwachajr) [inline hc/api/tests/test_pause.py]: maintain consistence in coding style
assert response.status_code == 200 or self.assertEqual(response.status_code, 405)
stick to one style of assertion

Comment (andela-landia) [reply hc/api/tests/test_pause.py]: Agreed preferably `self.assertEqual(response.status_code, 405)`

File: hc/api/tests/test_admin.py [hunk 19]
```diff
@@ -17,3 +20,6 @@ def test_it_shows_channel_list_with_pushbullet(self):
         ch.save()
 
         ### Assert for the push bullet
+        is_present = Channel.objects.filter(user=self.alice, kind='pushbullet', value="test-token").exists()
+        self.assertTrue(is_present, msg='pushbullet not present')
```
Comment (bmwachajr) [inline hc/api/tests/test_admin.py:24]: Should is_present instead be is_not_present as the message with respects to `msg`

File: hc/api/tests/test_list_checks.py (hunk 13 above)
Comment (bkiseu) [reply hc/api/tests/test_list_checks.py:35]: it's self.get. function within class that calls the self.client.get

File: hc/api/tests/test_admin.py (hunk 19 above)
Comment (bkiseu) [reply hc/api/tests/test_admin.py:24]: No...it's checking whether the pushbullet is present. the message is displayed when the test fails.

File: hc/api/tests/test_ping.py (hunk 16 above)
Comment (andela-landia) [reply hc/api/tests/test_ping.py]: @Bryoo kindly check on this

File: hc/api/tests/test_sendalerts.py (hunk 11 above)
Comment (andela-landia) [reply hc/api/tests/test_sendalerts.py:43]: @Bryoo This has not been implemented

Comment (bkiseu) [reply hc/api/tests/test_sendalerts.py:43]: the handle_many is implemented at the top so I worked on handle_one

File: hc/api/tests/test_ping.py (hunk 14 above)
Comment (bkiseu) [reply hc/api/tests/test_ping.py]: huh?

File: hc/api/tests/test_create_check.py (hunk 7 above)
Comment (bkiseu) [reply hc/api/tests/test_create_check.py:14]: This was not part of my test. The test was already there beforehand.

**Pull request merged**